lib/widgets/CustomButton/custom_button.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import 'package:classroom_app/styles/color/colors.dart';
+
+class CustomButton extends StatelessWidget {
+  const CustomButton({
+    super.key,
+    required this.buttonText,
+    required this.buttonBgColor,
+    this.buttonTextColor,
+    this.buttonBorderColor,
+  });
+
+  final String buttonText;
+  final Color buttonBgColor;
+  final Color? buttonTextColor;
+  final Color? buttonBorderColor;
+
+  @override
+  Widget build(BuildContext context) {
+    double screenWidth = MediaQuery.of(context).size.width;
+    return SizedBox(
+      height: 56,
+      width: screenWidth * 0.4,
+      child: ElevatedButton(
+        onPressed: () {},
+        style: ElevatedButton.styleFrom(
+          padding: EdgeInsets.symmetric(
+            horizontal: screenWidth * 0.08,
+            vertical: 14,
+          ),
+          backgroundColor: buttonBgColor,
+          side: buttonBorderColor == null
+              ? null
+              : BorderSide(color: buttonBorderColor!),
+          shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(5),
+          ),
+        ),
+        child: Text(
+          buttonText,
+          style: TextStyle(
+            color: buttonTextColor ?? AppColors.white,
+            fontSize: screenWidth * 0.04,
```

Comment:
Deriving font size from screenWidth can cause inconsistencies across different devices.
Use MediaQuery.textScaleFactor or fixed sizes for better control.

```suggestion
            fontSize: 16 * MediaQuery.textScaleFactorOf(context),
```

Reply:
Though, both are responsive in different contexts and both are useful to make responsive. The suggested one is, If the user has increased the text size in their system settings, the text will be scaled up based on the text scale factor. The second one is that when screen size changes font size will be adapted with the screen size.

I am changing it to the suggested one.

Reply:
_**'textScaleFactorOf' is deprecated and shouldn't be used. Use textScalerOf instead. Use of textScaleFactor was deprecated in preparation for the upcoming nonlinear text scaling support. This feature was deprecated after v3.12.0-2.0.pre.**_

I am using **_fontSize: MediaQuery.textScalerOf(context).scale(16.0),_**